### PR TITLE
Wordle scoring unit tests and bugfix

### PR DESCRIPTION
**Pull request by mthiemsAtCorvus**

### CODE REVIEW

File: birdle/test/birdle/game/score_test.exs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  birdle_test("close", "cheer", "GBYBB")
+  birdle_test("close", "cocas", "GYBBY")
+  birdle_test("close", "leave", "YBBBG")
+  birdle_test("abide", "speed", "BBYBY")
+  birdle_test("erase", "speed", "YBYYB")
+  birdle_test("steal", "speed", "GBGBB")
+  birdle_test("crepe", "speed", "BYGYB")
+  birdle_test("steal", "teeth", "YBGBB")
+  birdle_test("bonus", "bunny", "GYGBB")
+  birdle_test("bunny", "ninen", "YBGBB")
```
Comment (7hoenix) [inline birdle/test/birdle/game/score_test.exs:58]: I think these tests would be better with a description of why the test is necessary.

File: birdle/test/birdle/game/score_test.exs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        case color_letter do
+          "G" -> :green
+          "Y" -> :yellow
+          _ -> :black
+        end
+      end)
+    )
+  end
+
+  defmacro birdle_test(answer, guess, score) do
```
Comment (7hoenix) [inline birdle/test/birdle/game/score_test.exs:19]: 🎉

File: birdle/test/birdle/game/score_test.exs (hunk 1 above)
Comment (mthiemsAtCorvus) [reply birdle/test/birdle/game/score_test.exs:58]: Certainly true, but honestly I didn't think through the "why" in most of these. Many were just copied from examples of corner cases I found by googling. I made up a few more.
